plugins/modules/s3_logging.py

```diff
@@ -102,16 +107,28 @@ def verify_acls(connection, module, target_bucket):
         "Permission": "FULL_CONTROL",
     }
 
+    grant_present = False
     for grant in current_grants:
         if grant == required_grant:
-            return False
+            grant_present = True
+
+    # XNOR
+    if not (module.params.get("acl") ^ grant_present):
+        return False
```

Comment:
I might be missing a use-case but I think we can be as simple as:

```suggestion
    if module.params.get("acl") == grant_present:
        return False
```